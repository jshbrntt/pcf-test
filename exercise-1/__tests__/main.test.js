const { traverseGrid } = require("../src/main.js");

test("original test case", () => {
  const gridString = `
    HI  1  2  3   v4
    >9  10 11 v12 13
    ^14 15 16 17  <18
    `;
  console.log(gridString);
  const path = traverseGrid(gridString);
  console.log(path);
  expect(path).toMatchSnapshot();
});

test("grid with no control characters", () => {
  const gridString = `
      HI 1  2  3  4
      9  10 11 12 13
      14 15 16 17 18
      `;
  console.log(gridString);
  const path = traverseGrid(gridString);
  console.log(path);
  expect(path).toMatchSnapshot();
});

test("depth first walk", () => {
  const gridString = `
    vHI >1  v2  >3  v4
    9   10  11  12  13
    >14 ^15 >16 ^17 18
    `;
  console.log(gridString);
  const path = traverseGrid(gridString);
  console.log(path);
  expect(path).toMatchSnapshot();
});

test("breadth first walk", () => {
  const gridString = `
    HI  1  2  3   v4
    v9  10 11 12 <13
    >14 15 16 17  18
    `;
  console.log(gridString);
  const path = traverseGrid(gridString);
  console.log(path);
  expect(path).toMatchSnapshot();
});

test("spiral walk", () => {
  const gridString = `
    HI  a   b   c   d   e   f  vg
    >h  i   j   k   l   m  vn   o
    p  ^q   r   s   t   u  <v   w
    ^x  y   z   1   2   3   4  <5
    `;
  console.log(gridString);
  const path = traverseGrid(gridString);
  console.log(path);
  expect(path).toMatchSnapshot();
});

test("spiral walk v character", () => {
  const gridString = `
    HI  a   b   c   d   e   f  vg
    >h  i   j   k   l   m  vn   o
    p  ^q   r   s   t   u   v   w
    ^x  y   z   1   2   3  ^4  <5
    `;
  console.log(gridString);
  const path = traverseGrid(gridString);
  console.log(path);
  expect(path).toMatchSnapshot();
});

test("1x1 grid walk", () => {
  const gridString = `
    HI
    `;
  console.log(gridString);
  const path = traverseGrid(gridString);
  console.log(path);
  expect(path).toMatchSnapshot();
});

test("2x2 grid walk", () => {
  const gridString = `
    HI v1
    3  <2
    `;
  console.log(gridString);
  const path = traverseGrid(gridString);
  console.log(path);
  expect(path).toMatchSnapshot();
});

test("loop 2x2 grid walk", () => {
  const gridString = `
    HI v1
    ^3 <2
    `;
  console.log(gridString);
  const path = traverseGrid(gridString);
  console.log(path);
  expect(path).toMatchSnapshot();
});

test("reverse 2x2 grid walk", () => {
  const gridString = `
    vHI  1
    >3  ^2
    `;
  console.log(gridString);
  const path = traverseGrid(gridString);
  console.log(path);
  expect(path).toMatchSnapshot();
});

test("reverse loop 2x2 grid walk", () => {
  const gridString = `
    vHI <1
    >3  ^2
    `;
  console.log(gridString);
  const path = traverseGrid(gridString);
  console.log(path);
  expect(path).toMatchSnapshot();
});
