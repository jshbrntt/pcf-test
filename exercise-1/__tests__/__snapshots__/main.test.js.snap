// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`1x1 grid walk 1`] = `"HI"`;

exports[`2x2 grid walk 1`] = `"HI, 1, 2, 3"`;

exports[`breadth first walk 1`] = `"HI, 1, 2, 3, 4, 13, 12, 11, 10, 9, 14, 15, 16, 17, 18"`;

exports[`depth first walk 1`] = `"HI, 9, 14, 15, 10, 1, 2, 11, 16, 17, 12, 3, 4, 13, 18"`;

exports[`grid with no control characters 1`] = `"HI, 1, 2, 3, 4"`;

exports[`loop 2x2 grid walk 1`] = `"HI, 1, 2, 3, LOOP"`;

exports[`original test case 1`] = `"HI, 1, 2, 3, 4, 13, 18, 17, 16, 15, 14, 9, 10, 11, 12, LOOP"`;

exports[`reverse 2x2 grid walk 1`] = `"HI, 3, 2, 1"`;

exports[`reverse loop 2x2 grid walk 1`] = `"HI, 3, 2, 1, LOOP"`;

exports[`spiral walk 1`] = `"HI, a, b, c, d, e, f, g, o, w, 5, 4, 3, 2, 1, z, y, x, p, h, i, j, k, l, m, n, v, u, t, s, r, q, LOOP"`;

exports[`spiral walk v character 1`] = `"HI, a, b, c, d, e, f, g, o, w, 5, 4, v, n, LOOP"`;
